apps/functions/applications-migration/common/migrators/general-s51-advice-migration.js

```diff
@@ -0,0 +1,53 @@
+import { QueryTypes } from 'sequelize';
+import { SynapseDB } from '../synapse-db.js';
+import { migrateS51AdviceForCase } from './s51-advice-migration.js';
+import { ODW_GENERAL_CASE_REF } from '../constants.js';
+
+/**
+ * @param {import('@azure/functions').Logger} logger
+ * @param {number} offset
+ */
+export const migrateGeneralS51Advice = async (logger, offset = 0) => {
+	const countQuery = `
+		SELECT COUNT(*) AS generalS51AdviceCount
+		FROM [odw_curated_db].[dbo].[nsip_s51_advice]
+		WHERE caseReference = '${ODW_GENERAL_CASE_REF}';
+	`;
+	const [{ generalS51AdviceCount }] = await SynapseDB.query(countQuery, {
+		type: QueryTypes.SELECT
+	});
+	logger.info(`Number of General S51 Advice found in odw_curated_db: ${generalS51AdviceCount}`);
+
+	const batchSize = 10;
+	while (offset < generalS51AdviceCount) {
+		const paginatedGeneralS51AdviceQuery = `
+			SELECT * FROM [odw_curated_db].[dbo].[nsip_s51_advice]
+			WHERE caseReference = '${ODW_GENERAL_CASE_REF}'
+			ORDER BY nsip_s51_advice.adviceId
+			OFFSET ${offset} ROWS
+			FETCH NEXT ${batchSize} ROWS ONLY;
+		`;
+
+		try {
+			await migrateS51AdviceForCase(logger, ODW_GENERAL_CASE_REF, paginatedGeneralS51AdviceQuery);
+		} catch (error) {
+			logger.error(`Batch with offset ${offset} failed to migrate.`, error?.response?.body, error);
+			throw error;
+		}
+		offset += batchSize;
+	}
```

Comment:
the offset is a function parameter, as that way it lets you restart the process from a particular point if an advice migration fails mid-way through the process. Like I mentioned in my earlier comment though, not sure how useful the ability to do this will be in practice

Reply:
Ah I see, that makes sense but because `offset` is a `number` type, it gets passed by value so mutating it won't persist once the function exits. If I'm understanding that right. I think you'd need to do something like return the offset that it reached from the function.

Reply:
aye, the idea is that when the function is invoked via a http post request (as in `apps/functions/applications-migration/general-s51-advice-migration/index.js`), you can optionally include the offset param (which you'll know the value of from the error logs if something's gone wrong) and it'll kick off the process again from the point of failure.

Reply:
Ahh I see, so you get it from the error logs and set it manually? That sounds fine then 👍